homeassistant/util/temperature.py

```diff
@@ -9,3 +10,17 @@ def fahrenheit_to_celsius(fahrenheit: float) -> float:
 def celsius_to_fahrenheit(celsius: float) -> float:
     """Convert a Celsius temperature to Fahrenheit."""
     return celsius * 1.8 + 32.0
+
+
+def convert(temperature: float, from_unit: str, to_unit: str) -> (float, str):
+    """Convert a temperature from one unit to another."""
+    if from_unit not in [TEMP_CELSIUS, TEMP_FAHRENHEIT] or \
+        to_unit not in [TEMP_CELSIUS, TEMP_FAHRENHEIT] or \
+            to_unit == from_unit:
+        # If no need or unknown conversion, return temperature
+        return temperature, from_unit
```

Comment:
I was trying to follow the previous logic - which was if it was not a valid unit system, return the value without conversion - I was hoping sending a tuple back would allow components to verify that the conversion happened.

I agree raising an error would probably be best - but I didn't want to break too much existing logic.

So options: Return value/unit or raise error - which one?